### PR TITLE
Remove unnecessary null pointer checks #257

### PR DESCRIPTION
**Pull request by kubaroth**
Removed checks according to cpp guidelines linked in the original post.

### CODE REVIEW

File: openvdb/unittest/TestDenseSparseTools.cc [hunk 1]
```diff
@@ -14,7 +14,7 @@ class TestDenseSparseTools: public CppUnit::TestCase
 {
 public:
     virtual void setUp();
-    virtual void tearDown() { if (mDense) delete mDense;}
+    virtual void tearDown() {delete mDense;}
```
Comment (danrbailey) [inline openvdb/unittest/TestDenseSparseTools.cc]: Looks good, thanks. One minor comment, I'd expect spaces around the statement:

```{ delete mDense; }``` not ```{delete mDense;}```

I haven't looked through the cpp guidelines to check that this is written there, but that feels more consistent with the rest of the codebase.

Don't worry about the Houdini debug build check failing, that's failing elsewhere, but the DCO will need to be fixed before we can accept.

Comment (kubaroth) [reply openvdb/unittest/TestDenseSparseTools.cc]: Yeah, I wasn't sure about this and decided to wait for comments first. BTW has there been any discussion before to introduce a clang format tool etc to streamline that process?

Comment (danrbailey) [reply openvdb/unittest/TestDenseSparseTools.cc]: I believe we talked about it at one point, but it's a fairly time-consuming thing to introduce and we don't get _a lot_ of external contributions. As a result, our focus regarding infrastructure has been on improving other higher priority things - CMake, windows builds, CI, etc.

**Comment by danrbailey** [comment]
Merged, thanks!

**Pull request merged**